### PR TITLE
Update dot net version

### PR DESCRIPTION
**Pull request by avanikp**
# Description and Motivation 

Update to dotnet version 2.1

# Change type 

- [ ] Bug fix
- [ ] New feature
- [X] Enhancement
- [ ] Breaking change (breaks backward compatibility)

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/device-simulation-dotnet/344)

### CODE REVIEW

File: WebService/Startup.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@
 using Microsoft.Extensions.Logging;
 using Newtonsoft.Json;
 using ILogger = Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics.ILogger;
+using Microsoft.AspNetCore.Mvc;
```
Comment (peterfelts) [inline WebService/Startup.cs]: Do we know why this is needed?

Comment (avanikp) [reply WebService/Startup.cs]: This is needed for the changes in line 75 (as per documentation for net core 2.1 upgrade) -

services.AddMvc().AddControllersAsServices().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);

File: scripts/docker/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM microsoft/dotnet:2.0.3-runtime-jessie
+FROM microsoft/dotnet:2.1-aspnetcore-runtime
```
Comment (saixiaohui) [inline scripts/docker/Dockerfile]: Please change the maintainer to our team's alias

File: WebService/Startup.cs [hunk 3]
```diff
@@ -71,7 +72,7 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
             services.AddCors();
 
             // Add controllers as services so they'll be resolved.
-            services.AddMvc().AddControllersAsServices();
+            services.AddMvc().AddControllersAsServices().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
```
Comment (saixiaohui) [inline WebService/Startup.cs]: do we need this additional function call? what's the downside of not having it? what's the benefit for having it?

File: scripts/docker/Dockerfile (hunk 2 above)
Comment (avanikp) [reply scripts/docker/Dockerfile]: updated

File: WebService/Startup.cs (hunk 3 above)
Comment (avanikp) [reply WebService/Startup.cs]: I'm adding it as per documentation here to get the latest behavior: https://docs.microsoft.com/en-us/aspnet/core/mvc/compatibility-version?view=aspnetcore-2.1

Else the default will set compatibility version to 2.0

**Pull request merged**